src/commands/list-kinesis-streams.js

```diff
@@ -0,0 +1,227 @@
+	const resp = await Kinesis.describeStream({
+		StreamName: streamName
+	}).promise();
+
+	return {
+		streamName: streamName,
+		arn: resp.StreamDescription.StreamARN,
+		status: resp.StreamDescription.StreamStatus,
+		shards: resp.StreamDescription.Shards,
+		readMbPercentage: formatFloat(
+			(streamStat["GetRecords.BytesSum"] /
+				(resp.StreamDescription.Shards.length *
+					BYTES_OUTGOING_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		),
+		readRecordPercentage: formatFloat(
+			(streamStat["GetRecords.RecordsSum"] /
+				(resp.StreamDescription.Shards.length *
+					MAX_RECORDS_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		),
+		writeMbPercentage: formatFloat(
+			(streamStat.IncomingBytesSum /
+				(resp.StreamDescription.Shards.length *
+					BYTES_INCOMING_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		),
+		writeRecordPercentage: formatFloat(
+			(streamStat.IncomingRecordsSum /
+				(resp.StreamDescription.Shards.length *
+					MAX_RECORDS_PER_SHARD *
+					ONE_MIN_IN_SECONDS)) *
+				100
+		)
+	};
+};
+
+const getUsageMetrics = async (streamNames, region) => {
+	const CloudWatch = new AWS.CloudWatch({ region });
+	const metricNames = [
+		["IncomingBytes", "Sum"],
+		["IncomingRecords", "Sum"],
+		["GetRecords.Bytes", "Sum"],
+		["GetRecords.Records", "Sum"]
+	];
+
+	const oneMinuteAgo = new Date();
+	oneMinuteAgo.setMinutes(oneMinuteAgo.getMinutes() - 1);
+
+	const queries = _.flatMap(streamNames, streamName =>
+		metricNames.map(([metricName, stat]) => ({
+			Id:
+				metricName.replace(".", "").toLowerCase() +
+				uuid()
+					.replace(/-/g, "")
+					.substr(0, 5),
+			Label: `${streamName}:${metricName}:${stat}`,
+			MetricStat: {
+				Metric: {
+					Dimensions: [
+						{
+							Name: "StreamName",
+							Value: streamName
+						}
+					],
+					MetricName: metricName,
+					Namespace: "AWS/Kinesis"
+				},
+				Period: ONE_MIN_IN_SECONDS,
+				Stat: stat
+			},
+			ReturnData: true
+		}))
+	);
+
+	// each GetMetricData request can send as many as 100 queries
+	const chunks = _.chunk(queries, 100);
+	const promises = chunks.map(async metricDataQueries => {
+		const resp = await CloudWatch.getMetricData({
+			StartTime: oneMinuteAgo,
+			EndTime: new Date(),
+			MetricDataQueries: metricDataQueries
+		}).promise();
+
+		return resp.MetricDataResults.map(res => {
+			const [name, metricName] = res.Label.split(":");
+			const dataPoint = res.Values[0] || 0;
+			return {
+				name,
+				metricName: metricName + "Sum",
+				dataPoint
+			};
+		});
+	});
+
+	// array of {shardId, metricName, dataPoint}
+	const results = _.flatMap(await Promise.all(promises));
+	const byShardId = _.groupBy(results, res => res.name);
+	return _.mapValues(byShardId, shardMetrics => {
+		const kvp = shardMetrics.map(({ metricName, dataPoint }) => [
+			metricName,
+			dataPoint
+		]);
+		return _.fromPairs(kvp);
+	});
+};
+
+const show = streamsDetails => {
+	const table = new Table({
```

Comment:
these column names are too long, can we shorten them? e.g. `# of Shard` => `Shards`, `Avg Kinesis Read Capacity (MB/Records) %` => `Read utilization`, `Avg Kinesis Write Capacity (MB/Records) %`> `Write utilization`?

if you feel these need more explanation, then let's add some description at the bottom of the table

Reply:
Done